grails-app/controllers/org/pih/warehouse/api/StockMovementApiController.groovy

```diff
@@ -161,6 +167,108 @@ class StockMovementApiController {
+            }
+
+            String csv = new String(importFile.bytes)
+            def settings = [separatorChar: ',', skipLines: 1]
+            csv.toCsvReader(settings).eachLine { tokens ->
+                String requisitionItemId = tokens[0]
+                String lotNumber = tokens[1] ?: null
+                Date expirationDate = tokens[2] ? Constants.EXPIRATION_DATE_FORMATTER.parse(tokens[2]) : null
+                String binLocation = tokens[3] ?: null
+                Integer quantityPicked = tokens[4] ? tokens[4].toInteger() : null
+
+                if (!requisitionItemId || quantityPicked == null) {
+                    throw new IllegalArgumentException("Requisition item id and quantity picked are required")
+                }
+
+                if (lotNumber?.contains("E") && NumberUtils.isNumber(lotNumber)) {
+                    throw new IllegalArgumentException("Lot numbers must not be specified in scientific notation. " +
+                            "Please reformat field with Lot Number: \"${lotNumber}\" to a number format")
+                }
+
+                PickPageItem pickPageItem = stockMovement?.pickPage?.pickPageItems?.find { it.requisitionItem?.id == requisitionItemId }
+
+                if (!pickPageItem) {
+                    throw new IllegalArgumentException("Requisition item id: ${requisitionItemId} not found")
+                }
+
+                AvailableItem availableItem = pickPageItem.availableItems?.find {
+                    (binLocation ? it.binLocation?.name == binLocation : !it.binLocation) &&
+                            lotNumber == it.inventoryItem?.lotNumber && LocalDate.fromDateFields(expirationDate).equals(LocalDate.fromDateFields(it.inventoryItem?.expirationDate))
+                }
+
+                if (!availableItem) {
+                    throw new IllegalArgumentException("There is no item available with lot: ${lotNumber ?: ""}, expiration date: ${tokens[2] ?: ""} and bin: ${binLocation ?: ""}")
+                }
+
+                RequisitionItem requisitionItem = pickPageItem.requisitionItem?.modificationItem ?: pickPageItem.requisitionItem
+
+                pickPageItem.picklistItems.each {
+                    if (it.id) {
+                        it.quantity = 0
+                    }
+                }
+                pickPageItem.picklistItems.add(new PicklistItem(
+                        requisitionItem: requisitionItem,
+                        inventoryItem: availableItem.inventoryItem,
+                        binLocation: availableItem.binLocation,
+                        quantity: quantityPicked,
+                        sortOrder: pickPageItem.sortOrder
+                ))
+            }
+
+            stockMovementService.createOrUpdatePicklistItem(stockMovement)
+
+        } catch (Exception e) {
+            // FIXME The global error handler does not return JSON for multipart uploads
```

Comment:
Would this fix be very time consuming?

Reply:
it's copied from items import (on add items page) and I don't have any time left in the ticket